Tests failing on Python 3.13

Over on typing-extensions we run the typed-argument-parser tests daily on various Python versions. They started failing on Python 3.13 (python/typing_extensions#513):

```

https://github.com/python/typing_extensions/actions/runs/12172317648/job/33950775320

typed-argument-parser tests are failing on 3.13:

=========================== short test summary info ============================
FAILED tests/test_to_tap_class.py::test_subclasser_subparser[function-args_string_and_arg_to_expected_value7] - assert "error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')" in "usage: pytest [--foo] --arg_int ARG_INT [--arg_bool]\n              [--arg_list [ARG_LIST ...]] [-h]\n              {a,b} ...\npytest: error: argument {a,b}: invalid choice: 'X' (choose from a, b)\n"
 +  where "error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')" = str(SystemExit("error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')"))
FAILED tests/test_to_tap_class.py::test_subclasser_subparser[Class-args_string_and_arg_to_expected_value7] - assert "error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')" in "usage: pytest [--foo] --arg_int ARG_INT [--arg_bool]\n              [--arg_list [ARG_LIST ...]] [-h]\n              {a,b} ...\npytest: error: argument {a,b}: invalid choice: 'X' (choose from a, b)\n"
 +  where "error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')" = str(SystemExit("error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')"))
FAILED tests/test_to_tap_class.py::test_subclasser_subparser[_Args-args_string_and_arg_to_expected_value7] - assert "error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')" in "usage: pytest [--foo] --arg_int ARG_INT [--arg_bool]\n              [--arg_list [ARG_LIST ...]] [-h]\n              {a,b} ...\npytest: error: argument {a,b}: invalid choice: 'X' (choose from a, b)\n"
 +  where "error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')" = str(SystemExit("error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')"))
FAILED tests/test_to_tap_class.py::test_subclasser_subparser[class_or_function_3-args_string_and_arg_to_expected_value7] - assert "error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')" in "usage: pytest [--foo] --arg_int ARG_INT [--arg_bool]\n              [--arg_list [ARG_LIST ...]] [-h]\n              {a,b} ...\npytest: error: argument {a,b}: invalid choice: 'X' (choose from a, b)\n"
 +  where "error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')" = str(SystemExit("error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')"))
======================== 4 failed, 301 passed in 1.58s =========================
```

https://github.com/python/typing_extensions/actions/runs/12172317648/job/33950775320

I suspect it's related to some of the argparse changes in Python 3.13.1. (Run `git diff v3.13.0 v3.13.1 -- Lib/argparse.py` in the CPython repo; a fair amount of code changed.) I haven't looked into this further.

--- a/tests/test_to_tap_class.py
+++ b/tests/test_to_tap_class.py
@@ -266,7 +266,7 @@ def _test_subclasser(
 
     if isinstance(arg_to_expected_value, SystemExit):
         stderr = _test_raises_system_exit(tap, args_string)
-        assert str(arg_to_expected_value) in stderr
+        assert re.search(str(arg_to_expected_value), stderr)
     elif isinstance(arg_to_expected_value, BaseException):
         expected_exception = arg_to_expected_value.__class__
         expected_error_message = str(arg_to_expected_value) or None
@@ -471,7 +471,7 @@ def test_subclasser_complex_help_message(class_or_function_: Any):
         (
             "--arg_int 1 --baz X --foo b",
             SystemExit(
-                "error: argument {a,b}: invalid choice: 'X' (choose from 'a', 'b')"
+                r"error: argument \{a,b}: invalid choice: 'X' \(choose from '?a'?, '?b'?\)"
             ),
         ),
         (
@@ -480,7 +480,7 @@ def test_subclasser_complex_help_message(class_or_function_: Any):
         ),
         (
             "--arg_int 1 --foo b --baz A",
-            SystemExit("""error: argument --baz: Value for variable "baz" must be one of ['X', 'Y', 'Z']."""),
+            SystemExit(r"""error: argument --baz: Value for variable "baz" must be one of \['X', 'Y', 'Z']."""),
         ),
     ],
 )

